utils/polus-bfio-util/bfio/bfio.py

```diff
@@ -56,193 +85,746 @@ class BioReader():
+
+        Args:
+            ct (int,list): List of ints indicating the channels or timepoints to load
+                If None, returns a list of ints
+            axis (str): Must be 'c', 't'
+
+        Returns:
+            list: list of ints indicating the first and last index in the dimension
+        """
+        assert axis in 'CT'
         if not ct:
-            ct = [t for t in range(0,self._xyzct[axis])] # number of timepoints
+            # number of timepoints
+            ct = list(range(0, self._xyzct[axis]))
         else:
-            if np.any(np.greater_equal(self._xyzct[axis],ct)):
-                ValueError('At least one of the {}-indices was larger than largest index ({}).'.format(axis,self._xyzct[axis]-1))
-            elif np.any(np.less(0,ct)):
-                ValueError('At least one of the {}-indices was less than 0.'.format(axis))
-            elif len(ct)==0:
-                ValueError('At least one {}-index must be selected.'.format(axis))
+            assert np.any(np.greater(self._xyzct[axis], ct)),\
+                'At least one of the {}-indices was larger than largest index ({}).'.format(axis, self._xyzct[axis]-1)
+            assert np.any(np.less_equal(0, ct)),\
+                'At least one of the {}-indices was less than 0.'.format(axis)
+            assert len(ct) != 0,\
+                'At least one {}-index must be selected.'.format(axis)
         return ct
     
-    def read_image(self,X=None,Y=None,Z=None,C=None,T=None,series=None):
-        '''
-        image = BioReader.open()
-        
-        -Inputs-
-        X,Y,Z -     If None, then load all values in the given dimension. If a
-                    2-tuple, load values within the range.
-        C,T -       If None, then load all channels/timepoints, respectively. If a
-                    list or tuple, load indicated channels/timepoints.
-        series -    Placeholder, in the future will open multi-series images
-        
-        -Outputs-
-        image - a 5D numpy array with dimensions corresponding to (y,x,z,channel,timepoint)
-        
-        -Example-
-        # Import javabridge and start the vm
-        import javabridge
-        javabridge.start_vm(class_path=bioformats.JARS)
-        
-        # Path to bioformats supported image
-        image_path = 'path/to/image'
-        
-        # Create the BioReader object
-        bf = BioReader(image_path)
-        
-        # Load the full image
-        image = bf.read_image()
-        
-        # Only load the first 256x256 pixels, will still load all Z,C,T dimensions
-        image = bf.read_image(X=(0,256),Y=(0,256))
+    def pixel_type(self):
+        """pixel_type Get the pixel type
+
+        One of the following strings will be returned:
         
-        # Only load the second channel
-        image = bf.read_image(C=[1])
+        'uint8':  Unsigned 8-bit pixel type
+        'int8':   Signed 8-bit pixel type
```

Comment:
Same as before, do we want to support complex values?

Reply:
I'll include it in Version 1.4.0/2.4.0 when I do a restructuring of the code to accomodate Sunny's request to simplify the code structure.